### ISSUE TITLE
Error publishing pod update to trunk w/ v1.8.0

### ISSUE THREAD

**Issue by schayes04** [CocoaPods/CocoaPods#9176]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

### Command

```
/usr/local/bin/pod trunk push
```

### Report

* What did you do?
Attempted to push a new version of my CocoaPod to trunk after just upgrading to v1.8.0 of CocoaPods. Note I was able to successfully push to trunk again after downgrading back v1.7.5 so I believe it's a newly introduced issue.

* What did you expect to happen?
Successfully push updated version

* What happened instead?

### Stack

```
   CocoaPods : 1.8.0
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15 (19A573a)
       Xcode : 11.0 (11A420a)
         Git : git version 2.21.0 (Apple Git-122)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : master - git - https://github.com/CocoaPods/Specs.git @ 2e30f8b059308f9d079ea40712c26dbe89b2d75a
               mentalfaculty - git - https://github.com/mentalfaculty/Specs.git @ 3da88f5809fcce2ece9f374180a7428441b7853d
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.4.0
cocoapods-try         : 1.1.0
```

### Error

```
NoMethodError - undefined method `url' for nil:NilClass
/Library/Ruby/Gems/2.6.0/gems/cocoapods-trunk-1.4.0/lib/pod/command/trunk/push.rb:155:in `master_repo_url'
/Library/Ruby/Gems/2.6.0/gems/cocoapods-trunk-1.4.0/lib/pod/command/trunk/push.rb:121:in `validate_podspec'
/Library/Ruby/Gems/2.6.0/gems/cocoapods-trunk-1.4.0/lib/pod/command/trunk/push.rb:70:in `run'
/Library/Ruby/Gems/2.6.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.6.0/gems/cocoapods-1.8.0/lib/cocoapods/command.rb:52:in `run'
/Library/Ruby/Gems/2.6.0/gems/cocoapods-1.8.0/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

### Stack

```
   CocoaPods : 1.8.0
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15 (19A573a)
       Xcode : 11.0 (11A420a)
         Git : git version 2.21.0 (Apple Git-122)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : master - git - https://github.com/CocoaPods/Specs.git @ f6ae54c5de6f9ca264011750377969d0c2c954a8
               mentalfaculty - git - https://github.com/mentalfaculty/Specs.git @ 3da88f5809fcce2ece9f374180a7428441b7853d
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.4.0
cocoapods-try         : 1.1.0
```

**Comment by dnkoutso** [comment]
What does `pod repo list` show for you?

**Comment by dnkoutso** [comment]
cc @igor-makarov

**Comment by schayes04** [comment]
```
master
- Type: git (master)
- URL:  https://github.com/CocoaPods/Specs.git
- Path: /Users/shayes/.cocoapods/repos/master

privaterepo
- Type: git (master)
- URL:  https://github.com/privaterepo/Specs.git
- Path: /Users/shayes/.cocoapods/repos/privaterepo

2 repos
```

**Comment by dnkoutso** [comment]
Does `ls` of `/Users/shayes/.cocoapods/repos/master` print any results?

**Comment by dnkoutso** [comment]
I think it might be a bug in which `trunk` is not added to your repo list.

**Comment by schayes04** [comment]
This is what it looks like:

```
AlgoliaSearch.yml	Gemfile.lock		Specs
CocoaPods-version.yml	README.md		netlify.toml
Gemfile			Scripts
```

**Comment by dnkoutso** [comment]
That looks a bit odd to me. As if its a CDN repo but its not?

**Comment by dnkoutso** [comment]
Can you `pod repo remove master` and use 1.8.0 to publish?

**Comment by dnkoutso** [comment]
One more question, what are your `source` declarations in your `Podfile`, if any?

**Comment by dnkoutso** [comment]
It seems like as if the CDN repo was never created/initialized when you used 1.8.0.

Before you run `pod repo remove master` can you try a `pod install` _after_ you upgrade to 1.8.0 and then try the trunk command again?

**Comment by schayes04** [comment]
Yep that seems to have done the trick. Running the `pod install` before the `pod trunk push` resolved the issue

**Comment by dnkoutso** [comment]
@igor-makarov maybe we can improve that in `cocoapods-trunk` plugin with a minor patch fix to print out a correct error message or something?

**Comment by igor-makarov** [comment]
So what is actually happening? Trunk doesn't get created when running `pod trunk push`?

**Comment by JiaDingYi** [comment]
I have met the same problem, but it was in the travis ci, so I think the main problem for me is how to add the CDN repo in CI server.

**Comment by igor-makarov** [comment]
`pod repo add trunk 'https://cdn.cocoapods.org'`

**Comment by lqIphone** [comment]
gem install cocoapods-trunk   ,   https://rubygems.org/gems/cocoapods-trunk

**Comment by dnkoutso** [comment]
> So what is actually happening? Trunk doesn't get created when running pod trunk push?

@igor-makarov I think so yes.

**Comment by igor-makarov** [comment]
I've PR'd a fix: https://github.com/CocoaPods/cocoapods-trunk/pull/135

**Comment by dnkoutso** [comment]
Fix will be released with `cocoapods-trunk` 1.4.1 gem.

**Comment by dnkoutso** [comment]
Released with 1.4.1 `cocoapods-trunk` gem. CocoaPods automatically uses `>= 1.4.0` so new clients should receive the fixed version instead.
